app/src/main/java/com/codepath/apps/restclienttemplate/TweetsAdapter.java

```diff
@@ -145,6 +185,122 @@ public void bind(Tweet tweet) {
 
             tvBody.setTextColor(Color.BLACK);
             tvName.setTextColor(Color.BLACK);
+
+            tvRetweet.setText(String.valueOf(tweet.retweetCount));
+            tvFavorite.setText(String.valueOf(tweet.favoriteCount));
+
+            if (tweet.retweeted) {
+                ivRetweet.setImageResource(R.drawable.ic_vector_retweet);
+                ivRetweet.setTag(R.drawable.ic_vector_retweet);
+            } else {
+                ivRetweet.setImageResource(R.drawable.ic_vector_retweet_stroke);
+                ivRetweet.setTag(R.drawable.ic_vector_retweet_stroke);
+            }
+
+            if (tweet.favorited) {
+                ivFavorite.setImageResource(R.drawable.ic_vector_heart);
+                ivFavorite.setTag(R.drawable.ic_vector_heart);
+            } else {
+                ivFavorite.setImageResource(R.drawable.ic_vector_heart_stroke);
+                ivFavorite.setTag(R.drawable.ic_vector_heart_stroke);
+            }
+
+
+            ivRetweet.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View view) {
+                    Tweet tweet = tweets.get(getAdapterPosition());
+
+                    if ((int) ivRetweet.getTag() == R.drawable.ic_vector_retweet_stroke) {
```

Comment:
I think condition based on styling detail of a view is not ideal (what if you decide to change the icon or there's a bug when setting ivRetweet's tag?). I'd recommend checking if tweet.retweeted == true instead